From: Jonny Lamb <jonnylamb@gnome.org>
Date: Sat, 20 Mar 2010 23:47:08 +0000 (+0000)
Subject: main-window: only migrate butterfly logs if TPL is not enabled (bug #613437)
X-Git-Url: http://git.collabora.co.uk/?p=user%2Fjonny%2Fempathy.git;a=commitdiff_plain;h=c501014a8b6658c3be5c17f131954340131d32e7;hp=b81b3b92534ce3a0594e03315e3cdfe3b67fe1e3

main-window: only migrate butterfly logs if TPL is not enabled (bug #613437)

Signed-off-by: Jonny Lamb <jonnylamb@gnome.org>
---

diff --git a/src/Makefile.am b/src/Makefile.am
index a01b759..dc1af9a 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -110,13 +110,17 @@ empathy_handwritten_source = \
 	empathy-ft-manager.c empathy-ft-manager.h			\
 	empathy-invite-participant-dialog.c empathy-invite-participant-dialog.h \
 	empathy-main-window.c empathy-main-window.h			\
-	empathy-migrate-butterfly-logs.c empathy-migrate-butterfly-logs.h \
 	empathy-new-chatroom-dialog.c empathy-new-chatroom-dialog.h	\
 	empathy-preferences.c empathy-preferences.h			\
 	empathy-sidebar.c empathy-sidebar.h				\
 	empathy-status-icon.c empathy-status-icon.h			\
 	empathy.c
 
+if !ENABLE_TPL
+empathy_handwritten_source += \
+	empathy-migrate-butterfly-logs.c empathy-migrate-butterfly-logs.h
+endif
+
 empathy_SOURCES =							\
 	$(empathy_handwritten_source)					\
 	$(NULL)
diff --git a/src/empathy-main-window.c b/src/empathy-main-window.c
index fe9af9c..e4def87 100644
--- a/src/empathy-main-window.c
+++ b/src/empathy-main-window.c
@@ -63,7 +63,10 @@
 #include "empathy-chatrooms-window.h"
 #include "empathy-event-manager.h"
 #include "empathy-ft-manager.h"
+
+#ifndef ENABLE_TPL
 #include "empathy-migrate-butterfly-logs.h"
+#endif
 
 #define DEBUG_FLAG EMPATHY_DEBUG_OTHER
 #include <libempathy/empathy-debug.h>
@@ -121,8 +124,10 @@ typedef struct {
 	/* Actions that are enabled when there are connected accounts */
 	GList                  *actions_connected;
 
+#ifndef ENABLE_TPL
 	/* The idle event source to migrate butterfly's logs */
 	guint butterfly_log_migration_contact_added_id;
+#endif
 } EmpathyMainWindow;
 
 static EmpathyMainWindow *main_window = NULL;
@@ -1276,6 +1281,7 @@ account_manager_prepared_cb (GObject *source_object,
 	g_list_free (accounts);
 }
 
+#ifndef ENABLE_TPL
 static void
 main_window_contact_added_cb (EmpathyContactMonitor	*monitor,
 			      EmpathyContact		*contact,
@@ -1287,6 +1293,7 @@ main_window_contact_added_cb (EmpathyContactMonitor	*monitor,
 		window->butterfly_log_migration_contact_added_id = 0;
 	}
 }
+#endif
 
 GtkWidget *
 empathy_main_window_show (void)
@@ -1436,8 +1443,12 @@ empathy_main_window_show (void)
 							   EMPATHY_CONTACT_FEATURE_ALL);
 	g_signal_connect (monitor, "contact-presence-changed",
 			  G_CALLBACK (main_window_contact_presence_changed_cb), window);
+
+#ifndef ENABLE_TPL
 	window->butterfly_log_migration_contact_added_id =  g_signal_connect (monitor, "contact-added",
 			  G_CALLBACK (main_window_contact_added_cb), window);
+#endif
+
 	g_object_unref (list_iface);
 
 	gtk_widget_show (GTK_WIDGET (window->list_view));
